relay-server/src/actors/global_config.rs

```diff
@@ -271,3 +277,37 @@ impl Service for GlobalConfigService {
         });
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use relay_system::{Controller, Service, ShutdownMode};
+    use relay_test::mock_service;
+    use std::sync::Arc;
+    use std::time::Duration;
+
+    use crate::actors::global_config::{Get, GlobalConfigService};
+
+    /// Tests whether the service shuts down gracefully when the signal is
+    /// received, without requesting more global configs.
+    #[tokio::test(start_paused = true)]
+    async fn test_service_shutdown() {
+        for mode in &[ShutdownMode::Graceful, ShutdownMode::Immediate] {
+            shutdown_service_with_mode(*mode).await;
+        }
+    }
+
+    async fn shutdown_service_with_mode(mode: ShutdownMode) {
+        let (upstream, _) = mock_service("upstream", (), |&mut (), _| {});
+
+        Controller::start(Duration::from_secs(1));
+
+        let service = GlobalConfigService::new(Arc::default(), upstream).start();
+
+        assert!(service.send(Get).await.is_ok());
+
+        Controller::shutdown(mode);
+        tokio::time::advance(Duration::from_millis(200)).await;
```

Comment:
Call `tokio::time::sleep` here to simulate more accurate behavior. As [per docs](https://docs.rs/tokio/latest/tokio/time/fn.advance.html):

> this method will not wait for the sleep calls it advanced past to complete. If you want to do that, you should instead call [sleep](https://docs.rs/tokio/latest/tokio/time/fn.sleep.html) and rely on the runtime’s auto-advance feature.